### PR TITLE
ref(ts): Convert alerts to TypeScript

### PR DESCRIPTION
**Pull request by lynnagara**
Converts alerts components to Typescript. Also moves alertLink into the links group.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/alert.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import TextBlock from 'app/views/settings/components/text/textBlock';
 import isPropValid from '@emotion/is-prop-valid';
 import space from 'app/styles/space';
 
-const StyledInlineSvg = styled(InlineSvg)`
+const StyledInlineSvg = styled(InlineSvg)<{size?: string}>`
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/alert.tsx]: Isn't this required?

File: src/sentry/static/sentry/app/components/alert.tsx [hunk 2]
```diff
@@ -65,7 +65,17 @@ const StyledTextBlock = styled(TextBlock)`
   align-self: center;
 `;
 
-const Alert = ({type, icon, iconSize, children, className, ...props}) => {
+type AlertProps = {
+  type?: string;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/alert.tsx]: We can do better on the type

https://github.com/getsentry/sentry/pull/14703/files#diff-3d15e1f6a6441ac3fdf4f164638bcaefR12

File: src/sentry/static/sentry/app/components/alert.tsx [hunk 3]
```diff
@@ -65,7 +65,17 @@ const StyledTextBlock = styled(TextBlock)`
...
+  icon?: string;
+  iconSize?: string;
+  children: React.ReactNode;
+  className?: string;
+  alignTop?: boolean;
+  system?: boolean;
+};
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/alert.tsx]: It needs div element types too https://github.com/getsentry/sentry/pull/14703/files#diff-3d15e1f6a6441ac3fdf4f164638bcaefR19

File: src/sentry/static/sentry/app/components/alert.tsx [hunk 4]
```diff
@@ -54,7 +54,7 @@ const alertStyles = ({theme, type, system, alignTop}) => css`
   ${system && getSystemAlertColorStyles(theme.alert[type])};
 `;
 
-const AlertWrapper = styled('div', {shouldForwardProp: isPropValid})`
+const AlertWrapper: any = styled('div', {shouldForwardProp: isPropValid})`
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/alert.tsx]: Why is this `any`?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/alert.tsx]: I think you can prob type the `alertStyles` stuff like this: https://github.com/getsentry/sentry/pull/14703/files#diff-3d15e1f6a6441ac3fdf4f164638bcaefR21

File: src/sentry/static/sentry/app/components/alert.tsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/alert.tsx]: yeah, i wasn't able to get this to compile, will give it another shot later.

**Pull request closed without merging**